src/generic_search.cpp

```diff
@@ -36,15 +36,18 @@ namespace solver {
 
 std::size_t GetTuningIterationsMax()
 {
-    return Value(MIOPEN_DEBUG_TUNING_ITERATIONS_MAX{}, std::numeric_limits<std::size_t>::max());
+    return !IsUnset(MIOPEN_DEBUG_TUNING_ITERATIONS_MAX{})
+               ? Value(MIOPEN_DEBUG_TUNING_ITERATIONS_MAX{})
+               : std::numeric_limits<std::size_t>::max();
```

Comment:
For all of the cases in this file, I would have preferred setting the `default value` as opposed to these ternary conditionals.

Reply:
These environment variables are declared in a separate file. So setting the defaults there would separate those values from the logic of these functions. I think returning the values in this fashion is closer to the original writing of these functions and easier to read.

Reply:
@cderb The original code uses the notion of "default value", so it seems natural to continue using it here. If it requires moving the declaration to the header, then this is fine 🤝